proxy.go

```diff
@@ -40,11 +40,11 @@ type DialFunc func(ctx context.Context, isCONNECT bool, network, addr string) (c
 type Proxy interface {
 	// Handle handles a single connection, with in specified separately in case
 	// there's a buffered reader or something of that sort in use.
-	Handle(ctx context.Context, in io.Reader, conn net.Conn) error
+	Handle(in io.Reader, conn net.Conn) error
 
 	// Connect opens a CONNECT tunnel to the origin without requiring a CONNECT
 	// request to first be sent on conn. It will not reply with CONNECT OK.
-	Connect(ctx context.Context, in io.Reader, conn net.Conn, origin string) error
+	Connect(in io.Reader, conn net.Conn, origin string) error
```

Comment:
(This comment applies to the `Handle` method above as well.)

We can remove the use of `ctx` here because we have replaced all use of it with logic based on passing `filters.ConnectionState` around.  There is another common use-case for `context.Context` parameters which is to implement cancellation and deadlines.  However (1) deadlines were barely handled and cancellations not at all and (2) I didn't find any code in any dependent projects which relied on these methods to handle cancellation and deadlines.

Reply:
I see that we use it in flashlight in `client.go` in our socks5 config:

```
conf := &socks5.Config{
		HandleConnect: func(ctx context.Context, conn net.Conn, req *socks5.Request, replySuccess func(boundAddr net.Addr) error, replyError func(err error) error) error {
			op, ctx := ops.BeginWithNewBeam("proxy", ctx)
			defer op.End()

			host := fmt.Sprintf("%v:%v", req.DestAddr.IP, req.DestAddr.Port)
			addr, err := client.reverseDNS(host)
			if err != nil {
				return op.FailIf(log.Errorf("Error performing reverseDNS for %v: %v", host, err))
			}
			errOnReply := replySuccess(nil)
			if errOnReply != nil {
				return op.FailIf(log.Errorf("Unable to reply success to SOCKS5 client: %v", errOnReply))
			}
			return op.FailIf(client.proxy.Connect(ctx, req.BufConn, conn, addr))
		},
	}
```

And I see that we use it in dialing [here](https://github.com/getlantern/proxy/pull/50/files#diff-7a0eda0d43627b11b85f773a1b2cab7fff6b7b20e8dc65beb8d638aabc3c6c3bL95), so I don't think it's safe to discard it.

Reply:
Yes, the upstream dial timing out is the main case we're concerned about, as in office that can in fact hang.

Reply:
I like adding a `dialTimeout` or something like that.